src/Series.cpp

```diff
@@ -1223,28 +1236,8 @@ SeriesImpl::advance(
         *iteration.m_closed = Iteration::CloseStatus::Open;
     }
 
-    auto oldFlushLevel = IOHandler()->m_flushLevel;
-    IOHandler()->m_flushLevel = FlushLevel::UserFlush;
-    try
-    {
-        switch( series.m_iterationEncoding )
-        {
-            using IE = IterationEncoding;
-        case IE::groupBased:
-        case IE::variableBased:
-            flushGorVBased( begin, end );
-            break;
-        case IE::fileBased:
-            flushFileBased( begin, end );
-            break;
-        }
-        IOHandler()->m_flushLevel = oldFlushLevel;
-    }
-    catch( ... )
-    {
-        IOHandler()->m_flushLevel = oldFlushLevel;
-        throw;
-    }
+    flush_impl(
+        begin, end, FlushLevel::UserFlush, /* flushIOHandler = */ false );
```

Comment:
Why don't we flush the IOHandler at this point?
Needs an inline comment, this gets quite involved.

Reply:
I've added/clarified the comment now. Note that the logic of `advance` hasn't changed, it was just so far a slightly modified manual version of `flush_impl()`, but I have now added this workflow to `flush_impl()`.